### PR TITLE
implement and test string processing functionality

### PR DESCRIPTION
**Pull request by nathanbegbie**
this a one to one mapping of the functions implemented [in this ipython notebook](https://github.com/turnhub/covid19-nlu/blob/develop/data_prep/notebooks/turn_automation_logic.ipynb)
from python to elixir.

Note that I created sample CSVs to test the outputs against locally. I have left the code to process the CSV data in here as it is useful to test locally, with more sensitive data.
I have not included the CSVs in version control.

Additionally, we encountered some failures with the removal of emojis in the following libs:
* [Emojix](https://www.hex.pm/packages/emojix)
* [exmoji](https://www.hex.pm/packages/exmoji)
* [remove-emoji](https://www.hex.pm/packages/remove_emoji)

`remove_emoji` seemed to perform the best, but failed on certain values. 
A PR was opened [here](https://github.com/guanting112/elixir_remove_emoji/pull/1)

and have searched for other emoji libs [using hex](https://www.hex.pm/packages?_utf8=%E2%9C%93&search=emoji&sort=recent_downloads)

### CODE REVIEW

**Comment by bduvenhage** [comment]
@nathanbegbie That feature branch has been merged and the notebook is now at https://github.com/turnhub/covid19-nlu/blob/develop/data_prep/notebooks/turn_automation_logic.ipynb

**Comment by nathanbegbie** [comment]
Thanks @bduvenhage - have updated the description link

File: lib/peach.ex [hunk 1]
```diff
@@ -1,88 +1,98 @@
...
   @doc """
   Normalize text
   Unicode NFKC (Normalisation Form Compatibility Composition) normalisation.
   """
-  def normalise_text do
-    # return normalize('NFKC', text)
-    :to_do
+  def normalise_text(phrase) do
+    # https://hexdocs.pm/elixir/1.9.4/String.html#normalize/2
+    :unicode.characters_to_nfc_binary(phrase)
```
Comment (bduvenhage) [inline lib/peach.ex]: @nathanbegbie The reference implementation uses nfkc. k for compatible :-)
As far as I understand this is better for application like searching and text classification.
Is there a reason you used nfc?

Comment (nathanbegbie) [reply lib/peach.ex]: Gonna be honest, I was lazy and that's what the [Elixir doc](https://hexdocs.pm/elixir/1.9.4/String.html#normalize/2) said was available from the erlang library. But I see [here](http://erlang.org/doc/man/unicode.html#characters_to_nfkc_binary-1) that it is an option. Will make the change

File: test/peach_test.exs [hunk 2]
```diff
@@ -1,40 +1,213 @@
...
-    assert Peach.hello() == :world
-  end
-
+  @tag timeout: 600_000
   test "normalise_text" do
-    assert Peach.normalise_text() == :to_do
+    test_data = [
+      ["foo", "foo"],
+      ["\n", "\n"],
+      ["🚼", "🚼"]
```
Comment (nathanbegbie) [inline test/peach_test.exs:11]: @bduvenhage do you think you could provide some test examples here? Feel free to pop them in this comment and I can add them.

File: test/peach_test.exs [hunk 3]
```diff
@@ -1,40 +1,215 @@
...
-
+  @tag timeout: 600_000
   test "normalise_text" do
-    assert Peach.normalise_text() == :to_do
+    test_data = [
+      ["foo", "foo"],
+      ["\n", "\n"],
+      ["🚼", "🚼"],
+      [[500], "Ǵ"],
+      [[600], "ɘ"]
```
Comment (nathanbegbie) [inline test/peach_test.exs]: @bduvenhage do you think you could provide some test examples here for normalizing text with NFKC? Feel free to pop them in this comment and I can add them to the code.

Comment (bduvenhage) [reply test/peach_test.exs]: @nathanbegbie Please use: 
"＋－．～）｝"  ==>  "+-.~)}"
"１２３"  ==>  "123"
"ａｂｃＡＢＣ"  ==>  "abcABC"

More examples at https://towardsdatascience.com/difference-between-nfd-nfc-nfkd-and-nfkc-explained-with-python-code-e2631f96ae6c

**Comment by bduvenhage** [comment]
@nathanbegbie Can we go ahead and merge this PR?

File: test/peach_test.exs (hunk 3 above)
Comment (nathanbegbie) [reply test/peach_test.exs]: @bduvenhage can you add those examples yourself? Might be a good intro-to-elixir challenge :)

**Comment by nathanbegbie** [comment]
@bduvenhage can you add those test cases and then we should merge. I'm happy for you to merge the code, as long as the testing with the CSVs went well.

**Comment by bduvenhage** [comment]
@nathanbegbie CSV tests went we'll :-)

I've pushed a fix to https://github.com/turnhub/peach/tree/test-things-locally for remove_numbers and for regex unicode support.

Will add the NFKC test, merge your branches and then my updates.

**Pull request merged**